modules/Evse15118D20/charger/ISO15118_chargerImpl.cpp

```diff
@@ -89,6 +89,53 @@ convert_dynamic_values(const iso15118::message_20::datatypes::BPT_Dynamic_DC_CLR
             convert_from_optional<float>(in.min_v2x_energy_request)};
 }
 
+iso15118::message_20::datatypes::Parameter convert_parameter(const types::iso15118_vas::Parameter& parameter) {
+    iso15118::message_20::datatypes::Parameter out;
+    out.name = parameter.name;
+
+    if (parameter.value.bool_value.has_value()) {
+        out.value = parameter.value.bool_value.value();
+    } else if (parameter.value.int_value.has_value()) {
+        out.value = (int32_t)parameter.value.int_value.value();
+    } else if (parameter.value.short_value.has_value()) {
+        out.value = (int16_t)parameter.value.short_value.value();
+    } else if (parameter.value.byte_value.has_value()) {
+        out.value = (int8_t)parameter.value.byte_value.value();
+    } else if (parameter.value.rational_number.has_value()) {
+        out.value = dt::from_float(parameter.value.rational_number.value());
+    } else if (parameter.value.finite_string.has_value()) {
+        out.value = parameter.value.finite_string.value();
+    } else {
+        throw std::invalid_argument("Invalid ParameterValue in convert_parameter: " + parameter.name +
+                                    " has no value set");
+    }
+
+    return out;
+}
+
+iso15118::message_20::datatypes::ParameterSet
+convert_parameter_set(const types::iso15118_vas::ParameterSet& parameter_set) {
+    iso15118::message_20::datatypes::ParameterSet out;
+    out.id = parameter_set.set_id;
+
+    out.parameter.reserve(parameter_set.parameters.size());
+    for (const auto& parameter : parameter_set.parameters) {
+        out.parameter.push_back(convert_parameter(parameter));
+    }
+
+    return out;
+}
+
+std::vector<iso15118::message_20::datatypes::ParameterSet>
+convert_parameter_set_list(const std::vector<types::iso15118_vas::ParameterSet>& parameter_set_list) {
+    std::vector<iso15118::message_20::datatypes::ParameterSet> out;
```

Comment:
You can use `dt::` instead of `iso15118::message_20::datatypes`

Reply:
Will change 👍